File: Demo/Application/Features/PayPal - Vault/BraintreeDemoPayPalVaultViewController.m
#import "BraintreeDemoPayPalVaultViewController.h"
@import BraintreePayPal;

@implementation BraintreeDemoPayPalVaultViewController

- (UIView *)createPaymentButton {
    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
    [button setTitle:NSLocalizedString(@"PayPal Vault", nil) forState:UIControlStateNormal];
    [button setTitleColor:[UIColor blueColor] forState:UIControlStateNormal];
    [button setTitleColor:[UIColor colorWithRed:50.0/255 green:50.0/255 blue:255.0/255 alpha:1.0] forState:UIControlStateHighlighted];
    [button setTitleColor:[UIColor lightGrayColor] forState:UIControlStateDisabled];
    [button addTarget:self action:@selector(tappedPayPalVault:) forControlEvents:UIControlEventTouchUpInside];
    return button;
}

- (void)tappedPayPalVault:(UIButton *)sender {
    self.progressBlock(@"Tapped PayPal - vault using BTPayPalDriver");
    [sender setTitle:NSLocalizedString(@"Processing...", nil) forState:UIControlStateDisabled];
    [sender setEnabled:NO];

    BTPayPalDriver *driver = [[BTPayPalDriver alloc] initWithAPIClient:self.apiClient];
    BTPayPalVaultRequest *request = [[BTPayPalVaultRequest alloc] init];
    request.activeWindow = self.view.window;
    [driver tokenizePayPalAccountWithPayPalRequest:request completion:^(BTPayPalAccountNonce * _Nullable tokenizedPayPalCheckout, NSError * _Nullable error) {
        [sender setEnabled:YES];

        if (error) {
            self.progressBlock(error.localizedDescription);
        } else if (tokenizedPayPalCheckout) {
            self.completionBlock(tokenizedPayPalCheckout);
        } else {
            self.progressBlock(@"Canceled");
        }
    }];
}

@end

File: Demo/Application/Features/PayPal - Vault/BraintreeDemoPayPalVaultViewController.swift
import UIKit
import BraintreePayPal

class BraintreeDemoPayPalVaultViewController: BraintreeDemoPaymentButtonBaseViewController {
    
    private lazy var paypalClient = BTPayPalClient(apiClient: apiClient)
    
    override func createPaymentButton() -> UIView! {
        lazy var payPalVaultButton: UIButton = {
            let payPalVaultButton = UIButton(type: .system)
            payPalVaultButton.setTitle("PayPal Vault", for: .normal)
            payPalVaultButton.setTitleColor(.blue, for: .normal)
            payPalVaultButton.setTitleColor(.lightGray, for: .highlighted)
            payPalVaultButton.setTitleColor(.lightGray, for: .disabled)
            payPalVaultButton.addTarget(self, action: #selector(tappedPayPalVault), for: .touchUpInside)
            return payPalVaultButton
        }()
        
        return payPalVaultButton
    }
    
    @objc func tappedPayPalVault(_ sender: UIButton) {
        progressBlock("Tapped PayPal - Vault using BTPayPalClient")
        sender.setTitle("Processing...", for: .disabled)
        sender.isEnabled = false
        
        let request = BTPayPalVaultRequest()
        
        paypalClient.tokenize(request) { nonce, error in
            sender.isEnabled = true
            
            guard let nonce = nonce else {
                self.progressBlock(error?.localizedDescription)
                return
            }
            self.nonceStringCompletionBlock(nonce.nonce)
        }
    }
}
